server/src/repositories/person.repository.ts

```diff
@@ -252,16 +290,44 @@ export class PersonRepository implements IPersonRepository {
     return this.personRepository.save(entity);
   }
 
+  createMany(entities: Partial<PersonEntity>[]): Promise<PersonEntity[]> {
+    return this.personRepository.save(entities);
+  }
+
   async createFaces(entities: AssetFaceEntity[]): Promise<string[]> {
     const res = await this.assetFaceRepository.save(entities);
     return res.map((row) => row.id);
   }
 
+  async upsertFaces(assetId: string, entities: AssetFaceEntity[], sourceType?: string): Promise<string[]> {
+    return await this.assetFaceRepository.manager.transaction(async (manager) => {
+      const builder = manager.createQueryBuilder(AssetFaceEntity, 'asset_faces');
+      let query = builder.delete().where('assetId = :assetId', { assetId: assetId });
+
+      if (sourceType !== undefined) {
+        // eslint-disable-next-line unicorn/prefer-ternary
+        if (sourceType == null) {
+          query = query.andWhere({ sourceType: IsNull() });
+        } else {
+          query = query.andWhere('sourceType = :sourceType', { sourceType: sourceType });
+        }
+      }
```

Comment:
This could be flattened to `if` / `else if`.

Reply:
Done: https://github.com/immich-app/immich/compare/main...bugfest:immich:faces-from-metadata#diff-bdd10fc89e503edc0aa770142c760f507c6b6817b12b1b3a28df3e7d7145ca69R299-R303